### PR TITLE
Add chart

### PR DESCRIPTION
**Pull request by jacobtomlinson**
**Replaces the bulk of pangeo-data/pangeo#181.**

This PR is a work in progress and I'm keen on discussion and debate to help shape it. The primary goals here are to create a new helm chart for Pangeo which wraps up dependent charts (currently just [zero-to-jupyterhub](https://github.com/jupyterhub/zero-to-jupyterhub-k8s)) with extra kubernetes resources to tie everything together.

### Things implemented in this PR
- [x] Introduce the bare bones of a helm chart.
- [x] Move the dask-kubernetes RBAC config from #172 into the chart.
- [x] Update the GCE instructions to point to the new chart.
- [x] Indent current helm config under a `jupyterhub` key as per the [helm documentation for dependencies](https://docs.helm.sh/developing_charts/#chart-dependencies).
- [x] Add a `NOTES.txt` file to be displayed after an installation.
- [x] Create mechanism for deploying the chart repository at https://pangeo-data.github.io/helm-chart/

### Discussion topics

- I've linked the RBAC config option between the pangeo values and the jupyterhub values. This makes sense for config options which are always going to be the same, like RBAC. I've not used this approach before and would be interested in how people feel about it. I worry it may not be clear what is going on.
```yaml
# Setting the RBAC config options for the pangeo chart also sets them in the jupyterhub chart. 

# This means that 
---
rbac:
  enabled: true
---

# is equivalent to
---
rbac:
  enabled: true

jupyterhub:
  rbac:
    enabled: true
---
```
- I feel the `worker-template.yaml` file for dask-kubernetes would be better suited in the helm chart and exposed via a volume rather than bundled in the notebook Docker image. This would also allow customization of the workers from the chart config without having to build a new image.

### CODE REVIEW

**Comment by jacobtomlinson** [comment]
Zero to Jupyter Hub uses [chartpress](https://github.com/jupyterhub/chartpress) to publish their helm charts. We could also use that.

**Comment by rabernat** [comment]
Chartpress looks like a good tool. We should try follow the practices of the jupyter team as much as possible I think.

File: pangeo/templates/NOTES.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+It might take a few minutes for it to appear!
+
+
+Support
+-------
+
+Note that this is still an alpha release! If you have questions, feel free to
+  1. Come chat with us at https://groups.google.com/forum/#!forum/pangeo
+  2. File issues at https://github.com/pangeo-data/pangeo/issues
```
Comment (jhamman) [inline pangeo/templates/NOTES.txt]: I think you can drop (1). We haven't really been using it and have been encouraging discussion topics to be posted at: https://github.com/pangeo-data/pangeo/issues.

**Comment by jacobtomlinson** [comment]
I have now updated our Met Office helm chart to depend on this one and have successfully deployed it.

I'm going to merge this and tag a `v0.1.0` release. Then continue work on pangeo-data/pangeo#181 to actually make use of it there.

**Comment by mrocklin** [comment]
Huzzah!  I'm glad to see the explicit connection between the two
deployments :)

**Pull request merged**